test/e2e/must-gather/must_gather_suite_test.go

```diff
@@ -71,6 +71,7 @@ var _ = ginkgo.BeforeSuite(func() {
 		return
 	}
 	ginkgo.By("Setting up the cluster")
+	deployment.Deployer = deploy.NewForPlatform(configuration.Plat)
```

Comment:
thanks, this is a nice catch. I think I like more a fix like
```
$ git diff
diff --git a/test/e2e/must-gather/must_gather_suite_test.go b/test/e2e/must-gather/must_gather_suite_test.go
index 838cdc42..2da25052 100644
--- a/test/e2e/must-gather/must_gather_suite_test.go
+++ b/test/e2e/must-gather/must_gather_suite_test.go
@@ -62,12 +62,12 @@ var _ = ginkgo.BeforeSuite(func() {
        mustGatherTag = getStringValueFromEnv(envVarMustGatherTag, defaultMustGatherTag)
        ginkgo.By(fmt.Sprintf("Using must-gather image %q tag %q", mustGatherImage, mustGatherTag))
 
-       if _, ok := os.LookupEnv("E2E_NROP_INFRA_SETUP_SKIP"); ok {
-               ginkgo.By("Fetching up cluster data")
+       var err error
+       deployment, err = deploy.GetDeploymentWithSched(context.TODO())
+       gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
-               var err error
-               deployment, err = deploy.GetDeploymentWithSched(context.TODO())
-               gomega.Expect(err).ToNot(gomega.HaveOccurred())
+       if _, ok := os.LookupEnv("E2E_NROP_INFRA_SETUP_SKIP"); ok {
+               ginkgo.Expect(deployment.NroSchedObj).ToNot(gomega.BeNil(), "infra setup skip but Scheduler instance not found")
                return
        }
        ginkgo.By("Setting up the cluster")
```
Do you think the above could work?

Reply:
nice idea, thanks for raising! I updated GetDeploymentWithSched() a bit to initialize the deployer so even if setup is skipped it won't hurt. let me know if you meant something else please